query-engine/connector-test-kit-rs/query-engine-tests/tests/queries/batching/select_different_key_types.rs

```diff
@@ -0,0 +1,254 @@
+                  }"#,
+                r#"mutation entity {
+                    createOneDecimalEntity(data: { Id: 2, Text: "B" }) {
+                      Text
+                    }
+                }"#,
+            ] {
+                runner.query(mutation).await?.assert_success();
+            }
+            Ok(())
+        }
+    }
+
+    #[test_suite(schema(schema))]
+    mod datetime_in_schema {
+        fn schema() -> String {
+            let schema = indoc! {
+                r#"
+                model DateTimeEntity {
+                    #id(Id, DateTime, @id)
+                    Text String
+                }
+            "#
+            };
+
+            schema.to_owned()
+        }
+
+        #[connector_test]
+        async fn batch_of_two_distinct(runner: Runner) -> TestResult<()> {
+            create_test_data(&runner).await?;
+            assert_consistent_with_batch(
+                &runner,
+                r#"query { findUniqueDateTimeEntity(where: { Id: "2020-01-01T00:00:00Z" }){ Text }}"#,
+                r#"query { findUniqueDateTimeEntity(where: { Id: "2020-01-02T00:00:00Z" }){ Text }}"#,
+            )
+            .await
+        }
+
+        #[connector_test]
+        async fn batch_of_two_repeated(runner: Runner) -> TestResult<()> {
+            create_test_data(&runner).await?;
+            assert_consistent_with_batch(
+                &runner,
+                r#"query { findUniqueDateTimeEntity(where: { Id: "2020-01-01T00:00:00Z" }){ Text }}"#,
+                r#"query { findUniqueDateTimeEntity(where: { Id: "2020-01-01T00:00:00Z" }){ Text }}"#,
+            )
+            .await
+        }
+
+        async fn create_test_data(runner: &Runner) -> TestResult<()> {
+            for mutation in [
+                r#"mutation entity {
+                    createOneDateTimeEntity(data: { Id: "2020-01-01T00:00:00Z", Text: "A" }) {
+                      Text
+                    }
+                }"#,
+                r#"mutation entity {
+                    createOneDateTimeEntity(data: { Id: "2020-01-02T00:00:00Z", Text: "B" }) {
+                      Text
+                    }
+                }"#,
+            ] {
+                runner.query(mutation).await?.assert_success();
+            }
+            Ok(())
+        }
+    }
+
+    async fn assert_consistent_with_batch(runner: &Runner, query_a: &str, query_b: &str) -> TestResult<()> {
+        // These pairs of queries are run as batched and non-batched to verify that the
+        // batching logic returns the same results as the non-batched logic.
+        // The reason this is valuable is that the batching logic, unlike regular queries
+        // relies on a comparison operation implemented in our code, which is sensitive
+        // to differences in types of values.
+
+        let batch = runner
+            .batch(vec![query_a.to_owned(), query_b.to_owned()], false, None)
+            .await?
+            .into_data();
```

Comment:
Right now `into_data()` returns an empty vec if the response was a `Response::Error`, so the tests will pass if all queries fail. Do we want to assert that we are actually comparing the data?

Reply:
there's an assertion that checks whether it returned 2 rows

Reply:
ah right

Reply:
I'll add `assert_success`, it'll generate a better message